src/haxeparser/HaxeParser.hx

```diff
@@ -826,10 +949,17 @@ class HaxeParser extends hxparse.Parser<HaxeTokenSource, Token> implements hxpar
 			case [{tok:Arrow}, t2 = parseComplexType()]:
 				switch(t2) {
 					case TFunction(args,r):
-						TFunction(apush(args,t),r);
+						TFunction(aunshift(args,t),r);
```

Comment:
Hmm, what's the story here?

Reply:
`var f : (Int, Float) -> Int;` would print as `var f : (Float, Int) -> Int;` in unittest, so I looked it up and it's `t :: args` which means it should be unshift not push (as far as my understanding of ocaml goes)